### ISSUE TITLE
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136

### ISSUE THREAD

**Issue by Johnyboy1984** [mampfes/hacs_waste_collection_schedule#1850]
### I Have A Problem With:

The integration in general

### What's Your Problem

Entries that represent residual waste and other waste will no longer be generated.

### Source (if relevant)

Own Google ICS

### Logs

```Shell
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 17:08:30 (1 occurrences)
Last logged: 17:08:30

fetch failed for source ICS: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/ics.py", line 179, in fetch return self.fetch_url(self._url, self._params) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/ics.py", line 204, in fetch_url return self._convert(r.text) ^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/ics.py", line 215, in _convert dates = self._ics.convert(data) ^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py", line 39, in convert end_date = start_date.replace(year=start_date.year + 1) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ValueError: day is out of range for month
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: ics
      args:
        url: "https://calendar.google.com/.......
      customize:
        - type: Restmüll
          alias: Restmüll
          icon: mdi:trash-can
        - type: Gelbe Tonne
          alias: Altpapier und Gelber Sack
          icon: mdi:trash-can
          
  fetch_time: "04:00"
  day_switch_time: "10:00"

sensor:
  - platform: waste_collection_schedule
    name: Graue Tonne_date
    value_template: '{{value.date.strftime("%d.%m.%Y")}}'
    types:
      - Restmüll
      
  - platform: waste_collection_schedule
    name: Graue Tonne_collection
    value_template: "{{value.daysTo}}"
    types:
      - Restmüll

  - platform: waste_collection_schedule
    name: Gelbe Tonne_date
    value_template: '{{value.date.strftime("%d.%m.%Y")}}'
    types:
      - Altpapier und Gelber Sack
 
  - platform: waste_collection_schedule
    name: Gelbe Tonne_collection
    value_template: "{{value.daysTo}}"
    types:
      - Altpapier und Gelber Sack

  - platform: waste_collection_schedule
    name: AbfallNaechster
    count: 4
    value_template: '{{value.types|join(", ")}}|{{value.daysTo}}|{{value.date.strftime("%d.%m.%Y")}}|{{value.date.strftime("%a")}}'

#Senor Benachrichtigung, wie viele Tage

  - platform: waste_collection_schedule
    name: AbfallNotify
    details_format: 'upcoming'
    value_template: '{{value.daysTo}}'

  - platform: waste_collection_schedule
    name: AbfallNotifyTyp
    details_format: "upcoming"
    value_template: '{{value.types}}'
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by karflyer** [comment]
The error should occur for all users today.
Part of the error message reads: 'ValueError: day is out of range for month'. I suspect this is due to the fact that this year is a leap year and therefore February has 29 days instead of 28. Everything will probably be fine again tomorrow.

**Comment by 5ila5** [comment]
Leap year error fixed in #1855. But should work normally today.

ICS helper calculated an end date by replacing the year with year +1 in the current date, which throws an error as the 2025-02-29 does not exist.
